### PR TITLE
[cypress] Add tests for delete several gift cards

### PR DESCRIPTION
**Pull request by karolm-saleor**
Autotest for SALEOR-1009
Delete several Gift Cards

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [CLOUD-190 Switch to new flow after agreement](https://app.clickup.com/t/2549495/CLOUD-190)

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 1]
```diff
@@ -92,4 +94,57 @@ describe("As an admin I want to update gift card", () => {
         });
     },
   );
+
+  it(
+    "should be able to delete several gift cards. TC: SALEOR_1009",
+    { tags: ["@giftCard", "@allEnv", "@stable"] },
+    () => {
+      const name = `${startsWith}${faker.datatype.number()}`;
```
Comment (michalina-graczyk) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:102]: what is the name for? please add more detailed name

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 2]
```diff
@@ -92,4 +94,57 @@ describe("As an admin I want to update gift card", () => {
...
+      const name2 = `${startsWith}${faker.datatype.number()}`;
```
Comment (michalina-graczyk) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:103]: Please don't use name2, it doesn't say much

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 3]
```diff
@@ -25,7 +27,7 @@ describe("As an admin I want to update gift card", () => {
     cy.clearSessionData().loginUserViaRequest();
   });
 
-  it(
+  it.skip(
```
Comment (michalina-graczyk) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:30]: Why is it skipped?

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 4]
```diff
@@ -50,7 +52,7 @@ describe("As an admin I want to update gift card", () => {
     },
   );
 
-  it(
+  it.skip(
```
Comment (michalina-graczyk) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:55]: same here

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 5]
```diff
@@ -92,4 +94,57 @@ describe("As an admin I want to update gift card", () => {
...
+            .should("be.visible")
+            .get(BUTTON_SELECTORS.deleteItemsButton)
+            .first()
+            .click()
+            .get(GIFT_CARD_UPDATE.consentCheckbox)
+            .click()
+            .get(BUTTON_SELECTORS.submit)
+            .click()
+            .get(ASSIGN_ELEMENTS_SELECTORS.checkbox)
+            .should("not.be.visible");
```
Comment (michalina-graczyk) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:146]: I am afraid that this assertion might be not enough, can we check via API if there is null in response?

**Review by michalina-graczyk — CHANGES_REQUESTED** [review]

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 6]
```diff
@@ -92,4 +94,57 @@ describe("As an admin I want to update gift card", () => {
...
+            tag: name2,
+            amount: 7,
+            currency: "THB",
+          });
+        })
+        .then(() => {
+          cy.visit(urlList.giftCards)
+            .get(PRODUCTS_LIST.showFiltersButton)
+            .click()
+            .get(PRODUCTS_LIST.filters.filterBy.currency)
```
Comment (eCzerniak) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:121]: I don't think that using filter by currency is a good way. Someone else can use it and how we will now that we are removing proper giftCards?

**Review by eCzerniak — CHANGES_REQUESTED** [review]

File: cypress/elements/catalog/products/products-list.js [hunk 7]
```diff
@@ -23,13 +23,15 @@ export const PRODUCTS_LIST = {
   showFiltersButton: '[data-test-id="show-filters-button"]',
   filters: {
     filterOption: '[data-test-id="filter-option"]',
+    filterOptionField: "[class^=FilterOptionField]",
```
Comment (karola312) [inline cypress/elements/catalog/products/products-list.js:26]: If its possible avoid using classes, it would be better if you add and use data-test-id

**Review by karola312 — CHANGES_REQUESTED** [review]

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 8]
```diff
@@ -92,4 +94,57 @@ describe("As an admin I want to update gift card", () => {
...
+            .get(BUTTON_SELECTORS.submit)
+            .click()
+            .get(GIFT_CARD_UPDATE.giftCardRow)
+            .eq(1)
+            .should("be.visible")
+            .get(ASSIGN_ELEMENTS_SELECTORS.checkbox)
+            .first()
+            .check()
+            .should("be.checked")
+            .get("tr")
```
[the remark this answers is not in the document]
Comment (michalina-graczyk) [reply cypress/e2e/catalog/giftCards/updatingGiftCards.js:135]: Definitely we shouldn't use 'tr'

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 9]
```diff
@@ -92,4 +95,53 @@ describe("As an admin I want to update gift card", () => {
...
+      let giftCard01hash;
+      let giftCard02hash;
+
+      createGiftCard({
+        tag: giftCard01,
+        amount: 3,
+        currency: "THB",
+      })
+        .then(hash => {
+          cy.log(hash.id);
```
Comment (szczecha) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:114]: I think that you can remove cy.log

File: cypress/e2e/catalog/giftCards/updatingGiftCards.js [hunk 10]
```diff
@@ -92,4 +95,53 @@ describe("As an admin I want to update gift card", () => {
...
+          giftCard01hash = hash.id;
+          createGiftCard({
+            tag: giftCard02,
+            amount: 7,
+            currency: "THB",
+          });
+        })
+        .then(hash2 => {
+          cy.log(hash2.id);
```
Comment (szczecha) [inline cypress/e2e/catalog/giftCards/updatingGiftCards.js:123]: I think that you can remove cy.log

**Review by szczecha — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

**Review by karola312 — APPROVED** [review]

**Review by michalina-graczyk — APPROVED** [review]

**Pull request merged**